### PR TITLE
chore: Add documentation for Azure OpenAI streaming

### PR DESCRIPTION
**Pull request by ZhongpinWang**
## Context

AI/gen-ai-hub-sdk-js-backlog#175.

Add documentation for Azure OpenAI streaming in `foundation-models`.

## Definition of Done

- ~[ ] Code is tested (Unit, E2E)~
- ~[ ] Error handling created / updated & covered by the tests above~
- [X] Documentation updated
- ~[ ] (Optional) Aligned changes with the Java SDK~
- ~[ ] (Optional) Release notes updated~

### CODE REVIEW

**Review by deekshas8 — APPROVED** [review]
It reads well and concise to me. I am not a fan of 5 levels of #, but I also don't have any other idea at the moment.

File: packages/foundation-models/README.md [hunk 1]
```diff
@@ -145,6 +147,82 @@ logger.info(
...
+const finishReason = response.getFinishReason();
+const tokenUsage = response.getTokenUsage();
+
+console.log(`Finish reason: ${finishReason}\n`);
+console.log(`Token usage: ${JSON.stringify(tokenUsage)}\n`);
+```
+
+##### Streaming the Delta Content
+
+The client provides a helper method to extract delta content and stream string directly.
```
Comment (KavithaSiva) [inline packages/foundation-models/README.md:181]: [suggestion] I wouldn't understand that the delta content contains the actual LLM output we are interested in, without knowing the structure with which OpenAI responds.

Should we consider also giving an example of how a data chunk looks like?

File: packages/foundation-models/README.md [hunk 2]
```diff
@@ -145,6 +147,82 @@ logger.info(
...
+
+```ts
+for await (const chunk of response.stream.toContentStream()) {
+  console.log(chunk); // will log the delta content
+}
+```
+
+Each chunk will be a defined string containing the delta content.
+Set `choiceIndex` parameter for `toContentStream()` method to stream a specific choice.
```
Comment (KavithaSiva) [inline packages/foundation-models/README.md:190]: [suggestion] Add that by default it prints the delta content of the first choice.

**Review by KavithaSiva — COMMENTED** [review]
Mostly looks good, just few minor suggestions :)

File: packages/foundation-models/README.md (hunk 1 above)
Comment (ZhongpinWang) [reply packages/foundation-models/README.md:181]: Normally, `data` should be typed. This will be fixed once we update the client with `2024-10-21` spec.

**Pull request merged**